rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -924,53 +924,20 @@ void RSocketStateMachine::writeNewStream(
   }
 }
 
-void RSocketStateMachine::writeRequestN(StreamId streamId, uint32_t n) {
-  outputFrameOrEnqueue(Frame_REQUEST_N(streamId, n));
+void RSocketStateMachine::writeRequestN(Frame_REQUEST_N&& frame) {
```

Comment:
Why do we need these specific function? We already call `outputFrameOrEnqueue(std::move(frame))` for each and every one of them. So why don't we just remove them all?

Reply:
I didn't want to expose `outputFrameOrEnqueue()` just yet, as I'm not sure if that's desirable.  Writing a `REQUEST_N` frame just sends the frame, but writing a `REQUEST_STREAM` frame has to do more bookkeeping within the `RSocketStateMachine`.  It'd be confusing to see why some callers did `outputFrameOrEnqueue(Frame_REQUEST_N{})` and some did `writeNewStream(...)`.

For now I'd rather keep all these methods as separate, to reduce confusion.  I agree that having one function simply "wrapping" another isn't great though.